trio/_ssl.py

```diff
@@ -331,18 +342,18 @@ class SSLStream(Stream, metaclass=Final):
     # SSLListener.__init__, and maybe the open_ssl_over_tcp_* helpers.
     def __init__(
         self,
-        transport_stream,
-        ssl_context,
+        transport_stream: Stream,
+        ssl_context: _stdlib_ssl.SSLContext,
         *,
-        server_hostname=None,
-        server_side=False,
-        https_compatible=False,
-    ):
+        server_hostname: str | None = None,
+        server_side: bool = False,
+        https_compatible: bool = False,
+    ) -> None:
```

Comment:
nit: inconsequential `-> None`. We ought to create a style guide somewhere for typing, and/or add a check to #2744 that removes/adds it.

Reply:
It is not inconsequential, I am fairly certain that `verify_types.json` changes for the better when it's there

Reply:
Are you referring to https://github.com/python-trio/trio/pull/2740#discussion_r1285180160 ?
I removed the `-> None` without any impact on `verify_types.json`

Reply:
Just bumping on this!

Reply:
https://mypy.readthedocs.io/en/stable/config_file.html#confval-local_partial_types